app/uk/gov/hmrc/transitmovementsrouter/controllers/MessagesController.scala

```diff
@@ -164,4 +170,45 @@ class MessagesController @Inject() (
     case _                     => EitherT.leftT(PresentationError.badRequestError(s"${messageType.code} is not valid for requests"))
   }
 
+  private def updateMessageStatus(
+    sdesResponse: SdesNotificationItem,
+    messageStatus: MessageStatus
+  )(implicit
+    hc: HeaderCarrier
+  ) = {
+    val (movementId, messageId) = extractMovementMessageId(sdesResponse)
+    (for {
+      persistenceResponse <- persistenceConnector
+        .patchMessageStatus(movementId, messageId, MessageUpdate(messageStatus))
+        .asPresentation
+
+    } yield persistenceResponse).fold[Result](
+      _ => Ok,
+      _ => Ok
+    )
+  }
+
+  private def extractMovementMessageId(sdesResponse: SdesNotificationItem): (MovementId, MessageId) =
+    ConversationId(
+      UUID.fromString(
+        sdesResponse.conversationId.get.value
+      )
+    ).toMovementAndMessageId
+
+  def handleSdesResponse() =
+    Action.async(parse.json) {
+      implicit request =>
+        implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromRequest(request)
+        parseAndLogSdesResponse(request.body) match {
+          case Left(presentationError: PresentationError) => Future.successful(Status(presentationError.code.statusCode)(Json.toJson(presentationError)))
+          case Right(sdesResponse) =>
+            sdesResponse.notification match {
+              case SdesNotification.FileProcessed =>
+                updateMessageStatus(sdesResponse, MessageStatus.Success)
+              case SdesNotification.FileProcessingFailure =>
+                updateMessageStatus(sdesResponse, MessageStatus.Failed)
```

Comment:
The result of this shouldn't be ignored.

Unlike Upscan, where it only needs to tell us about a file and that's it, it doesn't care after that, SDES is actually part of the message flow so should expect a 500, which signals it to retry later. 

There are just returning `Ok`, even if our call fails we're telling SDES that everything is fine, when it's not. I do understand that for Upscan we do swallow (most) errors, but for SDES, we want it to retry if we can't update the status.

Take this comment with the one that Sandeep made above. `updateMessageStatus(...)` should be part of a for comprehension and if it fails, we tell SDES that with an internal server error.

Reply:
In the ticket it is mentioned as we should always be returning a 200 unless the message is malformed and I followed the same here.

If it needs to changed then fine to take it up, please confirm

Reply:
I added a comment to the ticket, but to summarise:

> If something fails (as in, internal server error, e.g. could not connect to the database), **we should return a 500 to SDES to prompt it to retry** rather than ignore it and return a 2xx error.

Reply:
Added for this scenario